### ISSUE TITLE
MitoHiFi cannot find FASTA file in the directory where it is being run

### ISSUE THREAD

**Issue by Mpersicae** [marcelauliano/MitoHiFi#104]
Hello, 

I am running MitoHiFi on an insect genome assembly using singularity on an SGE grid. I first built the tool using:

`singularity build mitohifi.sif docker://ghcr.io/marcelauliano/mitohifi:master`

I then use the following parameters: 

```
singularity exec mitohifi.sif mitohifi.py \
-c genome.fasta \
-f reference.fasta \
-g reference.gbk \
-t 4
```

Where 'genome.fasta' is an assembly of my target genome, and 'reference.fasta' and 'reference.gbk' are pulled from a reference mitochondrial genome on GenBank. I am running the program in the same directory as the three files. I keep getting the same Error:

```
INFO:    Using cached SIF image
INFO:    Converting SIF file to temporary sandbox...
WARNING: underlay of /etc/localtime required more than 50 (109) bind mounts
WARNING: user: lookup groupid 10513: no such file or directory
2024-10-02 16:00:25 [INFO] Welcome to MitoHifi v2. Starting pipeline...
Traceback (most recent call last):
  File "/opt/MitoHiFi/src/mitohifi.py", line 566, in <module>
    main()
  File "/opt/MitoHiFi/src/mitohifi.py", line 94, in main
    rel_mito_len = getMitoLength.get_mito_length(args.f)
  File "/opt/MitoHiFi/src/getMitoLength.py", line 18, in get_mito_length
    num_sequences = len(list(SeqIO.parse(mito_file, "fasta")))
  File "/usr/local/lib/python3.6/dist-packages/Bio/SeqIO/__init__.py", line 605, in parse
    return iterator_generator(handle)
  File "/usr/local/lib/python3.6/dist-packages/Bio/SeqIO/FastaIO.py", line 183, in __init__
    super().__init__(source, mode="t", fmt="Fasta")
  File "/usr/local/lib/python3.6/dist-packages/Bio/SeqIO/Interfaces.py", line 48, in __init__
    self.stream = open(source, "r" + mode)
FileNotFoundError: [Errno 2] No such file or directory: 'reference.fasta'
INFO:    Cleaning up image...
```

It is not exactly clear to me what is going on. I see that the program cannot find the reference fasta, but I am not sure if it recognizes the other files in the same directory. I have tried providing the entire file path for all three files but I get the same error regardless, always referring to reference.fasta. Any ideas what might be happening? 

Thanks, 
Yonathan

**Comment by Mpersicae** [comment]
Update:

I tried working with the test data and I also could not get it to run. I downloaded the GenBank references using:

`singularity exec mitohifi.sif findMitoReference.py --species "Deilephila porcellus" --outfolder . --min_length 14000`

This downloaded NC_057303.1.fasta and NC_057303.1.gbk into my home directory. I then tried to run MitoHiFi on the test data:

`singularity exec mitohifi.sif mitohifi.py -r MitoHiFi/tests/ilDeiPorc1.reads.100.fa -f NC_057303.1.fasta -g NC_057303.1.gb -t 4 -o 5`

Here is the output:

`INFO:    Converting SIF file to temporary sandbox...
WARNING: underlay of /etc/localtime required more than 50 (109) bind mounts
WARNING: user: lookup groupid 10513: no such file or directory
2024-10-03 13:58:38 [INFO] Welcome to MitoHifi v2. Starting pipeline...
2024-10-03 13:58:38 [INFO] Length of related mitogenome is: 15312 bp
2024-10-03 13:58:38 [INFO] Number of genes on related mitogenome: 37
2024-10-03 13:58:38 [INFO] Running MitoHifi pipeline in reads mode...
2024-10-03 13:58:38 [INFO] 1. First we map your Pacbio HiFi reads to the close-related mitogenome
2024-10-03 13:58:38 [INFO] minimap2 -t 4 --secondary=no -ax map-hifi NC_057303.1.fasta MitoHiFi/tests/ilDeiPorc1.reads.100.fa | samtools view -@ 4 -b -F4 -F 0x800 -o reads.HiFiMapped.bam
2024-10-03 13:58:38 [INFO] 2. Now we filter out any mapped reads that are larger than the reference mitogenome to avoid NUMTS
2024-10-03 13:58:38 [INFO] 2.1 First we convert the mapped reads from BAM to FASTA format:
2024-10-03 13:58:38 [INFO] samtools fasta reads.HiFiMapped.bam > gbk.HiFiMapped.bam.fasta
2024-10-03 13:58:38 [INFO] Total number of mapped reads: 0
2024-10-03 13:58:38 [INFO] 2.2 Then we filter reads that are larger than 15312 bp
2024-10-03 13:58:38 [INFO] Number of filtered reads: 0
2024-10-03 13:58:38 [INFO] 3. Now let's run hifiasm to assemble the mapped and filtered reads!
2024-10-03 13:58:38 [INFO] hifiasm --primary -t 4 -f 0 -o gbk.HiFiMapped.bam.filtered.assembled gbk.HiFiMapped.bam.filtered.fasta
No gbk.HiFiMapped.bam.filtered.assembled.[a/p]_ctg.gfa file(s).
            An error may have occurred when assembling reads with HiFiasm.
`

This produced the following files:
gbk.HiFiMapped.bam.fasta
gbk.HiFiMapped.bam.filtered.assembled.ec.bin
gbk.HiFiMapped.bam.filtered.assembled.ovlp.reverse.bin
gbk.HiFiMapped.bam.filtered.assembled.ovlp.source.bin
gbk.HiFiMapped.bam.filtered.fasta
hifiasm.log
reads.HiFiMapped.bam

When I downloaded the NCBI data, the files were placed in my home directory, not the working directory with the mitohifi.sif file, which made me suspect this is an issue with singularity not being able to locate files, rather than with MitoHiFi. So, I moved the SIF to my home directory and tried re-running the test data from the home directory using the same singularity command, but I got the exact same error and set of output files. 

I also tried re-running my data after moving everything to the home directory, but I got the same error I pasted into my original question.
